modules/apps/account/account-test/src/testIntegration/java/com/liferay/account/internal/upgrade/v2_11_2/test/RoleResourceUpgradeProcessTest.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.account.internal.upgrade.v2_11_2.test;
+
+import com.liferay.account.model.AccountEntry;
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.kernel.dao.orm.EntityCache;
+import com.liferay.portal.kernel.dao.orm.FinderCache;
+import com.liferay.portal.kernel.model.ResourceAction;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.model.ResourcePermission;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.service.ResourceActionLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.test.rule.DataGuard;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Joao Cordeiro
+ */
+@DataGuard(scope = DataGuard.Scope.NONE)
+@RunWith(Arquillian.class)
+public class RoleResourceUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@After
+	public void tearDown() throws Exception {
+		_resourcePermissionLocalService.deleteResourcePermission(
+			_resourcePermission);
+	}
+
+	@Test
+	public void testRoleResourceUpgradeProcess() throws Exception {
+		ResourceAction resourceAction =
+			_resourceActionLocalService.addResourceAction(
+				AccountEntry.class.getName(), "DEACTIVATE",
+				RandomTestUtil.randomLong());
+
+		int roleId = RandomTestUtil.randomInt();
+
+		_resourcePermissionLocalService.addResourcePermission(
+			TestPropsValues.getCompanyId(), AccountEntry.class.getName(),
+			ResourceConstants.SCOPE_COMPANY, String.valueOf(0), roleId,
+			ActionKeys.DELETE);
+
+		_runUpgrade();
+
+		Assert.assertNotNull(
+			_resourceActionLocalService.fetchResourceAction(
+				AccountEntry.class.getName(), ActionKeys.DEACTIVATE));
+
+		_resourcePermission =
+			_resourcePermissionLocalService.fetchResourcePermission(
+				TestPropsValues.getCompanyId(), AccountEntry.class.getName(),
+				ResourceConstants.SCOPE_COMPANY, String.valueOf(0), roleId);
+
+		Assert.assertTrue(
+			_resourcePermissionLocalService.hasActionId(
+				_resourcePermission, resourceAction));
+	}
```

Comment:
mmm... I don't get this... please have a look to the test `modules/apps/commerce/commerce-wish-list-test/src/testIntegration/java/com/liferay/commerce/wish/list/internal/upgrade/v1_2_1/test/CommerceWishListUpgradeProcessTest.java`
it should very similar... in that case we where replacing an action with other two but the meaning should be the same